pkg/plugins/manager/client/client.go

```diff
@@ -101,8 +102,11 @@ func (s *Service) CallResource(ctx context.Context, req *backend.CallResourceReq
 				removeConnectionHeaders(res.Headers)
 				removeHopByHopHeaders(res.Headers)
 				removeNonAllowedHeaders(res.Headers)
+			} else {
+				res.Headers = map[string][]string{}
 			}
 
+			proxyutil.SetProxyResponseHeaders(res.Headers)
```

Comment:
this change seems unrelated?

Reply:
Yes, discussed historically https://github.com/grafana/grafana/pull/67145#discussion_r1175537374, and devised to remove the middleware to spare me some refactoring :) This being a middleware it can easily be missed to be registered and re-introduce security vulnerability. 

You're probably right that I should keep this out from these changes 👍

Reply:
it's okay if it's intentional / tested